### DISCUSSION TITLE
Unexpected changes while using mobx-undecorate to migrate from mobx v5 to mobx v6

### DISCUSSION THREAD

**Discussion by muneerashaik** [mobxjs/mobx#3615] [Q&A]
We used `mobx-undecorate` to migrate from mobx v5 to mobx v6.

It helps us to make the migration process smooth, But we have encountered a few unexpected changes in our code base

1. It removed the `!` from our class variable `todos` , How can I stop making this change?
2. By default it uses `makeObservable`, Can we configure `makeAutoObservable` as default ?, If yes, how can we do this?

**Here is the sample code for reference**

   From:
  ```
  class TodoStore{
   @observable todos!:ObservableMap<string, TodoModel>

   constructor(){
      this.init()
   }

   @action.bound
   init(){
      this.todos = new ObservableMap()
   }
}
   ```
   
   To:
   ```
   class TodoStore {
   private todos: ObservableMap<string, TodoModel>

   constructor() {
      makeObservable<TodoStore, 'todos'>(this, {
         todos: observable,
         init: action.bound
      })
      this.init()
   }


   init() {
      this.todos = new ObservableMap()
   }
}
   ```
   
   **Our mobx version**
   ```
   "mobx": "6.7.0",
   "mobx-react": "7.6.0",
   ```
   used `npx mobx-undecorate` command to migrate
   
   [mweststrate](https://github.com/mweststrate), [urugator](https://github.com/urugator)

**Comment by urugator** [comment]
> It removed the ! from our class variable todos , How can I stop making this change?

Could be due to older jscodeshift version (or rather it's depedency `recast` https://github.com/benjamn/recast/issues/1012). 
If I will find some time, I will take a look it at the weekend.
However https://github.com/facebook/jscodeshift/issues/424 block us from upgrading to  >=0.12.0 .
This PR https://github.com/mobxjs/mobx/pull/3478 is blocked for the same reason. If anyone has an idea of how to workaround it, let me know.

EDIT: maybe we could try to bump only `reacast` via `resolutions` ...

> By default it uses makeObservable, Can we configure makeAutoObservable as default ?, If yes, how can we do this?

Not at the moment. How should it work? Just change the call or also omit the overrides arg? Or only omit overrides that are not default? What about `@action`s? `makeAutoObservable` uses `autoAction` by default, which has a slightly different meaning.

**Reply by muneerashaik** [reply]
Thanks for the quick reply @urugator 

> By default it uses makeObservable, Can we configure makeAutoObservable as default ?, If yes, how can we do this?

It would be great to configure the `mobx-undecorate` changes like
- What to keep in the class constructor while migrating, `makeObservable` or `makeAutoObservable` with `overrides` and `options` etc

In our case, we wanted to use `makeAutoObservable` in our classes with no overrides

**Comment by mweststrate** [comment]
To avoid overengineering here and putting a lot of effort in a single use,
you could also try a simple find and replace to change it to
makeAutoObservable?

**Comment by urugator** [comment]
In case you don't wanna wait for PR (it can take some time as we have to deal with more or less, at least seamingly, unrelated problems):
```json
"resolutions": {      
   "recast": "^0.23.1"
},
```
should do the trick. Remove it once you're done with migration.
